radix-engine/src/transaction/transaction_executor.rs

```diff
@@ -336,14 +358,14 @@ where
                                 type_pointer,
                             ),
                             scrypto_encode(&BurnFungibleResourceEvent {
-                                amount: fee_summary.to_burn_amount(),
+                                amount: fee_destination.to_burn,
                             })
                             .unwrap(),
                         ));
 
                         // Finalize execution trace
                         let execution_trace =
```

Comment:
It would be nice if `execution_trace` was an `Option<TransactionExecutionTrace>` which we only have if the execution trace module is enabled. This allows clients such as the toolkit to be able to easily check if the receipt they have contains execution trace or not instead of checking if the execution trace is empty or not.

Reply:
Yes!